sinatra-example.rb

```diff
@@ -17,4 +19,73 @@ class SinatraExample < Sinatra::Base
     "slept for #{milliseconds} milliseconds"
   end
 
-end
\ No newline at end of file
+  get '/mem/alloc/:size_mb/?:leak?/?:once?' do
+    if params[:once] and params[:once]=1 and $leak_buffer
+      puts "Already has memory allocated, skipping"
+      return "Already has memory allocated, skipping"
+    end
+
+    size_mb = params[:size_mb].to_i
```

Comment:
Again, it is just a test app. 

If the value is wrong, it blows app with an exception, that is OK for us.